lib/ansible/modules/source_control/git.py

```diff
@@ -1019,6 +1019,14 @@ def main():
         dest = os.path.abspath(dest)
         if bare:
             gitconfig = os.path.join(dest, 'config')
+        elif os.path.isfile(os.path.join(dest, '.git')):
+            with open(os.path.join(dest, '.git'), 'r') as gitfile:
+                data = gitfile.read()
+            match = re.match('gitdir: (.+)$', data)
+            if match:
+                separate_git_dir = match.group(1)
+                os.environ['GIT_DIR'] = separate_git_dir
```

Comment:
@cmrd-senya I've got concerns about modifying globally accessible env vars, so after talking to a colleague he brought up that whenever you change the env var value you should revert it to its initial state after using it.
So I strongly recommend moving to CLI arg instead of messing with env vars.